src/search/move_picker.h

```diff
@@ -32,124 +32,102 @@ enum PickerPhase {
 
 class MovePicker {
 public:
-    constexpr void reset()
-    {
-        m_killerMoves.reset();
-        m_historyMoves.reset();
-        m_pvTable.reset();
-    }
-
-    constexpr PVTable& pvTable()
-    {
-        return m_pvTable;
-    }
-
-    constexpr const PVTable& pvTable() const
+    void setPhase(PickerPhase phase)
     {
-        return m_pvTable;
+        m_phase = phase;
     }
 
-    constexpr KillerMoves& killerMoves()
+    template<Player player> constexpr std::optional<movegen::Move> pickNextMove(SearchTable& searchTable, const BitBoard& board, movegen::ValidMoves& moves, uint8_t ply, std::optional<movegen::Move> ttMove = std::nullopt)
```

Comment:
Nitpick: All other methods have board as first argument, maybe keep it that way

Reply:
Done